tests/avif/avif2jpeg.c

```diff
@@ -0,0 +1,55 @@
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif /* HAVE_CONFIG_H */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "gd.h"
+
+/* A short program which converts a .avif file into a .jpg file - 
+  just to get a little practice with the basic functionality.
+ */
+
+int main(int argc, char **argv)
+{
+	gdImagePtr im;
+	FILE *in, *out;
+
+	if (argc != 3) {
```

Comment:
i guess this is more of a human util rather than an actual unittest.  these would prob be better under examples/.

but we'll still want an unittest for these APIs.  we'd want a small avif file to check into the tree with a reference png, and then use things like `gdTestImageCompareToFile` to make sure we can read it correctly.  and vice-versa.

that way we can run `make check` and have everything validate w/out user interaction.

Reply:
Makes sense. I have to write some more unit tests for this too...

Reply:
Now also writing the three tests I've seen for every other image format...

Reply:
im2im fails, even though the transformed images look the same to me when I write them to files... there are small changes, like red 0xFF => red 0xFE. Hmm. AVIF is lossy, so I'm not sure I'll get pixel perfection. I've noticed the image comparison is commented out in jpeg_im2im.c.

I'm going to try the approach you mention, which looks like what I see in jpeg/bug_github_18.c.

Reply:
My new tests are failing for the same reason as my old tests - R, G, or B values are often off by one. This even though I'm encoding at a quality that, according to `avif.h`, should be lossless:

`#define AVIF_QUANTIZER_LOSSLESS 0`

I wonder if @wantehchang might recognize what's going on here. Is there a missing bit of precision?

Some samples. (We're comparing `r1` to `r2`, `g1` to `g2`, etc.)

![Screen Shot 2021-02-15 at 6 02 43 PM](https://user-images.githubusercontent.com/29419498/108001163-b8809700-6fb9-11eb-8787-00fc8f400fe8.png)
![Screen Shot 2021-02-15 at 6 05 02 PM](https://user-images.githubusercontent.com/29419498/108001171-bcacb480-6fb9-11eb-903d-a68a4477e049.png)
![Screen Shot 2021-02-15 at 6 05 29 PM](https://user-images.githubusercontent.com/29419498/108001177-bfa7a500-6fb9-11eb-8458-c97c9f97cff2.png)

Reply:
I don't suppose libavif ever loses a bit of precision when convert between YUV and RGB? I don't spot anything obvious in, say, `avifImageYUV8ToRGB8Color`, but a lot of conversion does go on there...

Reply:
One of the tests where I see this is [compare_avif_to_png.c](https://github.com/libgd/libgd/blob/6851185549a4f6c91b8dad5d0d1fa3bfb48ef887/tests/avif/compare_avif_to_png.c).

Reply:
Fixed it. Tests pass. Thanks @wantehchang for the tip about `AVIF_MATRIX_COEFFICIENTS_IDENTITY`.